src/sources/external_input.ml

```diff
@@ -156,6 +156,9 @@ object (self)
 
   method stype = Source.Fallible
 
+  (* Whether we should read headers. *)
+  val mutable has_headers = true
```

Comment:
`need_headers`?

Reply:
You also need a mutex to protect changing/reading its value since it's gonna be accessed from different threads:
```
val mutable need_headers = true
val need_headers_m

method get_need_headers =
  Tutils.mutexify (fun () -> need_headers) ()

method set_need_headers =
  Tutils.mutexify (fun v -> need_headers <- v)
```

Reply:
Might be able to find better methods names but you see the idea..

Reply:
From the point of view of the source `has_header` indicates whether the stream has headers.

Reply:
I don't really see when there are multiple threads running here, there's only one duppy thread no?

Reply:
Nope, duppy has a model with multiple threads picking up pending tasks.